newIDE/app/src/ProjectCreation/LocalExamples.js

```diff
@@ -54,11 +55,14 @@ export const showGameFileCreationError = (
 
 export default class LocalExamples extends Component<Props, State> {
   state = {
-    outputPath: findEmptyPath(
-      path && app
-        ? path.join(app.getPath('documents'), 'GDevelop projects')
-        : ''
-    ),
+    outputPath:
+      this.props.pathProvider.getState() === ''
```

Comment:
Then the question is "sure I moved outputPath to a prop, and stored it in the parent, but how do I initialize it??". Two possibilities:
1. Ideally, because it's the parent that is the owner of the state, it's the parent responsibility to initialize it. So you would move this code up to the parent component.
2. This being said in this case the parent is common between the electron-app and the browser app. So as an exception, we might want to have the parent keep `outputPath` empty first. Then in the children, we add a `componentDidMount` (React lifecycle function). Inside, we do `if (!this.props.outputPath) this.props.onChangeOutputPath(...)` with ... being the logic to set the output path.